File: app/src/main/java/cedule/app/data/entities/Task.java
package cedule.app.data.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

import cedule.app.data.entities.Category;

@Entity(tableName = "tasks", foreignKeys = @ForeignKey(
        entity = Category.class,
        parentColumns = "id",
        childColumns = "category",
        onUpdate = 5,
        onDelete = 4
))
public class Task {
    @PrimaryKey
    public Integer id;

    public Integer createTime;

    public String title;

    public Integer category;

    public Long startDate;

    public Integer startTime;

    @ColumnInfo(defaultValue = "0")
    public Integer isDone;

    public Integer isNotify;

    public String note;
}

File: app/src/main/java/cedule/app/data/entities/Task.kt
package cedule.app.data.entities;

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "tasks",
    foreignKeys = [
        ForeignKey(
            entity = Category::class,
            parentColumns = ["id"],
            childColumns = ["category"],
            onUpdate = ForeignKey.CASCADE,
            onDelete = ForeignKey.SET_DEFAULT
        )
    ]
)
data class Task(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Int? = null,

    @ColumnInfo(name = "createTime", defaultValue = "CURRENT_TIMESTAMP")
    val createTime: Int? = null,

    @ColumnInfo(name = "title")
    val title: String? = null,

    @ColumnInfo(name = "category")
    var category: Int? = null,

    @ColumnInfo(name = "startDate")
    val startDate: Long? = null,

    @ColumnInfo(name = "startTime")
    val startTime: Int? = null,

    @ColumnInfo(name = "isDone", defaultValue = "0")
    val isDone: Boolean = false,

    @ColumnInfo(name = "isNotify", defaultValue = "0")
    val isNotify: Boolean = false,

    @ColumnInfo(name = "note")
    val note: String? = null
)